torch/distributions/transforms.py

```diff
@@ -972,6 +973,23 @@ def _inverse(self, y):
         return y.tril(-1) + y.diagonal(dim1=-2, dim2=-1).log().diag_embed()
 
 
+class PositiveDefiniteTransform(LowerCholeskyTransform):
```

Comment:
Please avoid inheriting from `LowerCholeskyTransform` here. Instead either own an instance or use the global instance.  I think something like this should work:
```python
class PositiveDefiniteConstraint(Transform):
    def __init__(self, cache_size=0):
        super.__init__(cache_size=cache_size)
        self._lower_cholesky = LowerCholesky()
    def _call(self, x):
        x = self._lower_cholesky(x)
        return x @ x.mT
    def _inverse(self, y):
        y = torch.linalg.cholesky(y)
        return self._lower_cholesky.inv(y)
```

Reply:
just for my understanding, why is this prefered over the current approach?

Reply:
In practice, downstream users may dispatch on transforms via `singledispatch`, `multipledispatch`, or `isinstance`, which could yield incorrect behavior due to the false match.

In theory, the two classes do not satisfy a subtyping relation, an [is-a](https://en.wikipedia.org/wiki/Is-a) relationship: `PositiveDefiniteTransform` is not a special case of `LowerCholeskyTransform`.

(sorry I had tried to explain in my earlier review, but apparently forgot to save the comment)